src/compiler/scala/tools/nsc/plugins/Plugin.scala

```diff
@@ -157,38 +159,26 @@ object Plugin {
     ignoring: List[String],
     findPluginClassloader: (Seq[Path] => ClassLoader)): List[Try[AnyClass]] =
   {
-    // List[(jar, Try(descriptor))] in dir
-    def scan(d: Directory) =
-      d.files.toList sortBy (_.name) filter (Jar isJarOrZip _) map (j => (j, loadDescriptionFromJar(j)))
-
     type PDResults = List[Try[(PluginDescription, ScalaClassLoader)]]
 
-    // scan plugin dirs for jars containing plugins, ignoring dirs with none and other jars
-    val fromDirs: PDResults = dirs filter (_.isDirectory) flatMap { d =>
-      scan(d.toDirectory) collect {
-        case (j, Success(pd)) => Success((pd, findPluginClassloader(Seq(j))))
+    val fromLoaders = paths.map {path =>
+      val loader = findPluginClassloader(path)
+      val list: Option[URL] = loader.getResources(PluginXML).asScala.take(1).toList.headOption
```

Comment:
`Option(loader.getResource(PluginXML))` or is there a mysterious gotcha? Besides they're going to rename it to `Option.ofNullable`?

Reply:
@som-snytt Oh yeah, that sounds better. I got too attached to the existing loops.